### PR TITLE
Fix exceptions when parsing invalid BiggerDecimal 

### LINKED ISSUE #600: Exceptions when parsing invalid BiggerDecimal
```scala
scala> import io.circe.numbers.BiggerDecimal
import io.circe.numbers.BiggerDecimal

scala> BiggerDecimal.parseBiggerDecimal("123.")
java.lang.NumberFormatException: For input string: "123."
  at java.lang.NumberFormatException.forInputString(NumberFormatException.java:65)
  at java.lang.Integer.parseInt(Integer.java:580)
  at java.math.BigInteger.<init>(BigInteger.java:461)
  at java.math.BigInteger.<init>(BigInteger.java:597)
  at io.circe.numbers.BiggerDecimal$.parseBiggerDecimalUnsafe(BiggerDecimal.scala:345)
  at io.circe.numbers.BiggerDecimal$.parseBiggerDecimal(BiggerDecimal.scala:230)
  ... 42 elided

scala> BiggerDecimal.parseBiggerDecimal("123e")
java.lang.NumberFormatException: For input string: "123e"
  at java.lang.NumberFormatException.forInputString(NumberFormatException.java:65)
  at java.lang.Integer.parseInt(Integer.java:580)
  at java.math.BigInteger.<init>(BigInteger.java:461)
  at java.math.BigInteger.<init>(BigInteger.java:597)
  at io.circe.numbers.BiggerDecimal$.parseBiggerDecimalUnsafe(BiggerDecimal.scala:345)
  at io.circe.numbers.BiggerDecimal$.parseBiggerDecimal(BiggerDecimal.scala:230)
  ... 42 elided

scala> BiggerDecimal.parseBiggerDecimal("123e-")
java.lang.NumberFormatException: Zero length BigInteger
  at java.math.BigInteger.<init>(BigInteger.java:430)
  at java.math.BigInteger.<init>(BigInteger.java:597)
  at io.circe.numbers.BiggerDecimal$.parseBiggerDecimalUnsafe(BiggerDecimal.scala:352)
  at io.circe.numbers.BiggerDecimal$.parseBiggerDecimal(BiggerDecimal.scala:230)
  ... 42 elided

```
Noticed this when trying to decode `"123."` into a `Long`, which in my case was invalid user input. Should either instead return `DecodingFailure`, or in the case of the first strip the `.` before parsing, but I'm learning towards the former.

This could be fixed by treating all `AFTER_*`as failure states. Can test this and put up a PR tomorrow

### PR DESCRIPTION
**Pull request by SamSchlegel**
Fixes #600 

/cc @travisbrown

### CODE REVIEW

File: modules/numbers/shared/src/main/scala/io/circe/numbers/BiggerDecimal.scala [hunk 1]
```diff
@@ -328,7 +328,7 @@ final object BiggerDecimal {
         i += 1
       }
 
-      if (state == FAILED) null else {
+      if (state == FAILED || state == AFTER_DOT || state == AFTER_E || state == AFTER_EXP_SIGN) null else {
```
Comment (SamSchlegel) [inline modules/numbers/shared/src/main/scala/io/circe/numbers/BiggerDecimal.scala]: Not sure what style you prefer here, so open to suggestions. Could also do a match, possibly with `@switch` but that's probably overkill

Comment (SamSchlegel) [inline modules/numbers/shared/src/main/scala/io/circe/numbers/BiggerDecimal.scala]: @travisbrown Not sure what style you prefer here, so open to suggestions. Could also do a match, possibly with `@switch` but that's probably overkill

Comment (travisbrown) [inline modules/numbers/shared/src/main/scala/io/circe/numbers/BiggerDecimal.scala]: This looks good to me.

**Comment by travisbrown** [comment]
Thanks! Will merge when green.

**Pull request merged**